### PR TITLE
Add notification queue

### PR DESCRIPTION
**Pull request by georgifarashev**
Notification queue which will be used by websocket controllers to receive notifications.

### CODE REVIEW

File: notifications/notification_queue.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+	"github.com/gofrs/uuid"
+
+	"github.com/Peripli/service-manager/pkg/types"
+)
+
+var _ NotificationQueue = &notificationQueue{}
+
+// NewNotificationQueue returns new NotificationQueue with specific size
+func NewNotificationQueue(size int) (NotificationQueue, error) {
```
Comment (KirilKabakchiev) [inline notifications/notification_queue.go]: Its better (according to the go guidelines) to return concrete type and not interface

File: notifications/interfaces.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+// ErrQueueClosed error stating that the queue is closed
+var ErrQueueClosed = errors.New("queue closed")
+
+// ErrQueueFull error stating that the queue is full
+var ErrQueueFull = errors.New("queue is full")
+
+// NotificationQueue is used for receiving notifications
+type NotificationQueue interface {
+	// Enqueue adds a new notification for processing. If queue is full ErrQueueFull is returned.
```
Comment (dpanayotov) [inline notifications/interfaces.go]: interface documentation should not be implementation-specific

File: notifications/interfaces.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	// It should not block or execute heavy operations.
+	Enqueue(notification *types.Notification) error
+
+	// Channel returns the go channel with received notifications which has to be processed.
+	// If error is returned this means that the NotificationQueue is no longer valid.
```
Comment (dpanayotov) [inline notifications/interfaces.go]: Why is this error needed? `notification, ok <-queue.Channel()` will return `false` if the channel is closed

File: notifications/interfaces.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	Channel() (<-chan *types.Notification, error)
+
+	// Close closes the queue.
+	// Any subsequent calls to Next or Enqueue will return ErrQueueClosed.
+	// Any subsequent calls to Close does nothing.
+	Close()
```
Comment (dpanayotov) [inline notifications/interfaces.go]: What does it mean to close a queue assuming that a queue might not be a channel?

File: notifications/notification_queue.go [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+type notificationQueue struct {
+	isClosed             bool
+	size                 int
+	notificationsChannel chan *types.Notification
+	id                   string
+}
+
+func (nq *notificationQueue) Enqueue(notification *types.Notification) error {
+	if nq.isClosed {
```
Comment (dpanayotov) [inline notifications/notification_queue.go]: can this be called by multiple goroutines? mutex may be needed

File: notifications/interfaces.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+	// ID returns unique queue identifier
+	ID() string
+}
+
+// Notificator is used for receiving notifications for SM events
```
Comment (dpanayotov) [inline notifications/interfaces.go]: not used in the PR

**Pull request closed without merging**